tests/test_jobs.py

```diff
@@ -89,6 +89,27 @@ def test_job_start_blocking(default_domino_client):
     job = default_domino_client.job_start_blocking(command="main.py")
     assert job["statuses"]["executionStatus"] == "Succeeded"
 
+@pytest.mark.skipif(not domino_is_reachable(), reason="No access to a live Domino deployment")
+def test_job_start_override_hardware_tier_id(default_domino_client):
+    """
+    Confirm that we can start a job using the v4 API and override the hardware tier id
+    """
+    new_project_name = f"project-{str(uuid.uuid4())}"
+    response = default_domino_client.project_create(new_project_name)
+    assert response.status_code == 200
+
+    project_list = default_domino_client.projects_list()
+    assert any(p['name'] == new_project_name for p in project_list)
+    created_project = next((p for p in project_list if p['name'] == new_project_name), None)
+
+    hardware_tiers = default_domino_client.hardware_tiers_list(created_project['id'])
+    non_default_hardware_tiers = (hardware_tier["isDefault"] == False for hardware_tier in hardware_tiers)
```

Comment:
This test ran and passed? The way I read the current line, `non_default_hardware_tiers` is being assigned a python generator that will produce a series of Boolean values. It looks like what you want a list comprehension instead, something like...
```suggestion
    non_default_hardware_tiers = [hwt for hwt in hardware_tiers if not hwt["isDefault"]]
```